### PR TITLE
Add custom layers 

### PR DESCRIPTION
**Pull request by Vimalakirti**
This PR added layers and related functions for RoPE, Gelu, and ReshapeTrans.
The python scripts used to edit the ONNX models were included under `scratch/`. (They might need to be refactored later.)

### CODE REVIEW

File: src/basic_block/matmul.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #![allow(non_snake_case)]
 #![allow(non_upper_case_globals)]
+#![allow(elided_named_lifetimes)]
```
Comment (ddkang) [inline src/basic_block/matmul.rs]: what's this?

Comment (Vimalakirti) [reply src/basic_block/matmul.rs]: It's used to bypass the elided name warning from line 13. The new commit fixed this warning.

File: scratch/replace_onnx_nodes/bert/replace_reshape_trans.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import onnx
```
Comment (liliatangxy) [inline scratch/replace_onnx_nodes/bert/replace_reshape_trans.py]: i think we don't need to have the `replace_onnx_nodes` directory

File: src/layer/rope.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+use crate::util;
+use ark_bn254::Fr;
+use ark_std::Zero;
+use ndarray::Dimension;
+use ndarray::{ArrayD, IxDyn};
+use tract_onnx::pb::tensor_proto::DataType;
+use tract_onnx::pb::AttributeProto;
+use tract_onnx::prelude::DatumType;
+
+fn get_rope_rotate_indices(input_shape: &Vec<usize>) -> ArrayD<Option<IxDyn>> {
```
Comment (liliatangxy) [inline src/layer/rope.rs:16]: can you add some comments on the logic?

Comment (Vimalakirti) [reply src/layer/rope.rs:16]: added

**Review by liliatangxy — APPROVED** [review]

**Review by ddkang — APPROVED** [review]

**Pull request merged**